### PR TITLE
New TBB API was used

### PR DESCRIPTION
**Pull request by averbukh**
# Description
Temporary TBB interface usage was replaced by brand-new public API

### CODE REVIEW

File: cpp/daal/src/threading/threading.cpp [hunk 1]
```diff
@@ -31,11 +31,15 @@
     #include <stdlib.h> // malloc and free
     #include <tbb/tbb.h>
```
Comment (alexey-katranov) [inline cpp/daal/src/threading/threading.cpp:32]: If tbb.h is included, there is no big sense to include other TBB headers.

Comment (averbukh) [reply cpp/daal/src/threading/threading.cpp:32]: I'll investigate it in the future. Probably, it's better to include particular headers of TBB instead, and get rid of tbb/tbb.h

Comment (alexey-katranov) [reply cpp/daal/src/threading/threading.cpp:32]: Great!

File: cpp/daal/src/threading/threading.cpp [hunk 2]
```diff
@@ -247,8 +251,8 @@ DAAL_EXPORT void _daal_del_mutex(void * mutexPtr)
 DAAL_EXPORT bool _daal_is_in_parallel()
 {
 #if defined(__DO_TBB_LAYER__)
-    #if defined(TBB_INTERFACE_VERSION) && TBB_INTERFACE_VERSION >= 12001
-    return tbb::detail::d1::task::current_execute_data() != nullptr;
+    #if defined(TBB_INTERFACE_VERSION) && TBB_INTERFACE_VERSION >= 12002
+    return tbb::task::current_context() != nullptr;
```
Comment (Vika-F) [inline cpp/daal/src/threading/threading.cpp:255]: Shouldn't it be something like:
    #if defined(TBB_INTERFACE_VERSION) && TBB_INTERFACE_VERSION == 12001
    return tbb::detail::d1::task::current_execute_data() != nullptr;
    #elif defined(TBB_INTERFACE_VERSION) && TBB_INTERFACE_VERSION >= 12002
    return tbb::task::current_context() != nullptr;
    #else
    return tbb::task::self().state() == tbb::task::executing;
    #endif

Comment (averbukh) [reply cpp/daal/src/threading/threading.cpp:255]: No, because there is no TBB with TBB_INTERFACE_VERSION == 12001. It's canceled.

**Pull request merged**